examples/file_body.hpp

```diff
@@ -79,7 +79,11 @@ struct file_body
         boost::tribool
         operator()(resume_context&&, error_code&, Write&& write)
         {
-            buf_len_ = std::min(size_ - offset_, sizeof(buf_));
+            if(size_ - offset_ < sizeof(buf_))
+                buf_len_ = static_cast<std::size_t>(
+                    size_ - offset_);
+            else
+                buf_len_ = sizeof(buf_);
```

Comment:
Would `buf_len_ = static_cast<std::size_t>(std::min(size_ - offset_, sizeof(buf_)));` be more efficient, or does upcasting `sizeof(buf_)` only to downcast it back make it not worth it?